src/main/kotlin/fi/metatavu/vp/usermanagement/workevents/WorkEventRepository.kt

```diff
@@ -110,6 +110,36 @@ class WorkEventRepository : AbstractRepository<WorkEventEntity, UUID>() {
         ).firstResult<WorkEventEntity>().awaitSuspending()
     }
 
+    /**
+     * Finds latest shift ending event
+     *
+     * @return shift ending event
+     */
+    suspend fun findLatestShiftEndingEvent(): WorkEventEntity? {
+        val sb = StringBuilder()
+        val parameters = Parameters.with("time", OffsetDateTime.now().minusHours(5))
+        addCondition(sb, "workShift.endedAt is NULL and time < :time");
+
+        return queryWithCount(
+            find(
+                sb.toString(),
+                Sort.descending("time"),
+                parameters
+        ), 0, 1).first.firstOrNull()
```

Comment:
Also add parameter names if possible
```suggestion
        return queryWithCount(
            find(sb.toString(), Sort.descending("time"), parameters),
            0,
            1
        ).first.firstOrNull()
```